lib/authlogic/session/callbacks.rb

```diff
@@ -98,7 +98,20 @@ def #{method}
 
         def save_record(alternate_record = nil)
           r = alternate_record || record
-          r.save_without_session_maintenance(:validate => false) if r && r.changed? && !r.readonly?
+          return unless r
+
+          ensure_valid_new_record(r) if r.new_record?
+
+          if r.errors.blank? && r.changed? && !r.readonly?
+            r.save_without_session_maintenance(:validate => false)
+          else
+            r
+          end
```

Comment:
@jaredbeck @radar @binarylogic I tracked the `validate: false` option down to a PR that [added Rails 3.* support](https://github.com/binarylogic/authlogic/pull/244/files), claiming to fix a Rails 3 deprecation message. I bet it was [this one](https://github.com/rails/rails/blob/3-0-stable/activerecord/lib/active_record/validations.rb#L70), which was just a message signature change.

Is there ever a good reason to save the record without validating?